File: apps/api/src/modules/memories/memory.controller.ts
import { Controller, Get, Param, Patch, Delete, Body } from '@nestjs/common';
import { MemoryService } from './memory.service.js';

@Controller('memories')
export class MemoryController {
  constructor(private readonly memoryService: MemoryService) {
    this.getSnapshot = this.getSnapshot.bind(this);
    this.listAgentMemories = this.listAgentMemories.bind(this);
    this.updateImportance = this.updateImportance.bind(this);
    this.removeAgentMemory = this.removeAgentMemory.bind(this);
  }

  @Get('snapshot/:conversationId')
  async getSnapshot(@Param('conversationId') conversationId: string) {
    return this.memoryService.getSnapshot(conversationId);
  }

  @Get('agent/:agentId')
  async listAgentMemories(@Param('agentId') agentId: string) {
    return this.memoryService.listAgentMemories(agentId);
  }

  @Patch('agent/:id/importance')
  async updateImportance(@Param('id') id: string, @Body('importance') importance: number) {
    return this.memoryService.updateAgentMemoryImportance(id, importance);
  }

  @Delete('agent/:id')
  async removeAgentMemory(@Param('id') id: string) {
    return this.memoryService.removeAgentMemory(id);
  }
}

File: apps/server-java/src/main/java/com/haoyitec/agent/server/module/memory/MemoryController.java
package com.haoyitec.agent.server.module.memory;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/memories")
public class MemoryController {

    private final MemoryService memoryService;

    @GetMapping("/snapshot/{conversationId}")
    public Object getSnapshot(@PathVariable String conversationId) {
        return memoryService.getSnapshot(conversationId);
    }

    @GetMapping("/agent/{agentId}")
    public Object listAgentMemories(@PathVariable String agentId) {
        return memoryService.listAgentMemories(agentId);
    }

    @PatchMapping("/agent/{id}/importance")
    public Object updateImportance(@PathVariable String id, @RequestBody Map<String, Object> body) {
        Integer importance = body.get("importance") == null ? null : Integer.parseInt(String.valueOf(body.get("importance")));
        return memoryService.updateAgentMemoryImportance(id, importance);
    }

    @DeleteMapping("/agent/{id}")
    public Object removeAgentMemory(@PathVariable String id) {
        return memoryService.removeAgentMemory(id);
    }
}

